### DISCUSSION TITLE
How do I make Escape exit the menu and discard my selection?

### DISCUSSION THREAD

**Discussion by K4LCIFER** [marlonrichert/zsh-autocomplete#328] [Q&A] [answered]
## What do you want?

When scrolling through the command history it would be nice to have a quick way to escape from the search which would also clear the command text. I would propose hitting `<esc>` as an option.

## Why do you want this?

Currently its a little inconvenient, as if you're searching through the history and want to get out, you have to hit `<tab>` on an item and then delete the command text.

## Who else would benefit from this?

I think everyone. It would be a QOL improvement.

## Explain in detail how you would expect this to work
Given the following situation:
* You're scrolling through the history menu, and you decide that you want to delete the command and start over

When I perform the following steps:
1. Hit the `<esc>` key.

Then I expect the following to happen:
*  The history search is exited and the previously autocompleted command from the search is cleared.

**Comment by marlonrichert** [comment, marked as answer]
Just add this to your `.zshrc` file:
```zsh
bindkey -M menuselect -s '^[' '^G^_'
```
Note that this won't work correctly when you're in the middle of doing a multi-select.
